Marlin/src/core/serial.cpp

```diff
@@ -29,7 +29,11 @@ static const char errormagic[] PROGMEM = "Error:";
 static const char echomagic[]  PROGMEM = "echo:";
 
 #if NUM_SERIAL > 1
-  int8_t serial_port_index = SERIAL_PORT;
+  #if defined(ECHO_TO_FIRST_SERIAL_PORT)
+    int8_t serial_port_index = 0;
+  #else
+    int8_t serial_port_index = SERIAL_PORT;
+  #endif
```

Comment:
The value in `serial_port_index` is a zero-based index where 0 refers to `SERIAL_PORT`, 1 refers to `SERIAL_PORT_2`, etc. The value of `SERIAL_PORT` might be -1 (to indicate native USB) but `serial_port_index` will still be 0 when output is being directed to `SERIAL_PORT`.